src/main/java/com/mathworks/ci/actions/RunMatlabTestsAction.java

```diff
@@ -0,0 +1,127 @@
+package com.mathworks.ci.actions;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+
+import hudson.FilePath;
+
+import com.mathworks.ci.Utilities;
+import com.mathworks.ci.MatlabBuilderConstants;
+import com.mathworks.ci.MatlabExecutionException;
+import com.mathworks.ci.parameters.TestActionParameters;
+import com.mathworks.ci.utilities.MatlabCommandRunner;
+
+public class RunMatlabTestsAction {
+    private MatlabCommandRunner runner;
+    private TestActionParameters params;
+
+    public RunMatlabTestsAction(MatlabCommandRunner runner, TestActionParameters params) {
+        this.runner = runner;
+        this.params = params;
+    }
+
+    public RunMatlabTestsAction(TestActionParameters params) throws IOException, InterruptedException {
+        this(new MatlabCommandRunner(params), params);
+    }
+
+    public void run() throws IOException, InterruptedException, MatlabExecutionException {
+        // Copy in genscript
+        FilePath genScriptZip = runner.copyFileToTempFolder(
+                MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR,
+                "genscript.zip");
+        genScriptZip.unzip(runner.getTempFolder());
+
+        // Prepare the command
+        String command = MatlabBuilderConstants.TEST_RUNNER_SCRIPT;
+        command = command.replace("${TEMP_FOLDER}", runner.getTempFolder().getRemote());
+        command = command.replace("${PARAMS}", getParameterString());
+        
+        // Run the command
+        try {
+            runner.runMatlabCommand(command);
+        } catch (Exception e) {
+            this.params.getTaskListener()
+                .getLogger()
+                .println(e.getMessage());
+            throw(e);
+        }
+    }
+
+    private String singleQuotify(String in) {
+        return "'" + in + "'";
+    }
+
+    // Concatenate the input arguments, try to keep this function as
+    // readable as possible because it can get hairy.
+    private String getParameterString() {
+        // The final list to be concatted and returned
+        final List<String> inputArgsList = new ArrayList<String>();
+
+        inputArgsList.add("'Test'");
+
+        // Prepare source and test folder lists
+        String sourceFolders = null;
+        if (this.params.getSourceFolder() != null) {
+            sourceFolders = this.params.getSourceFolder().size() == 0
+                ? null
+                : Utilities.getCellArrayFrmList(this.params.getSourceFolder());
```

Comment:
Seems a little funny to drop the "o" in `From` in an already long method name 😆

Reply:
Oh yeah, I didn't end up touching the `Utilities` class but I can make a quick pass over it and normalize it to be more similar to the rest of the updated codebase.